### PR TITLE
net: Use type-safe mockable time for peer connection time

### PR DESCRIPTION
**Pull request by maflcko**
Benefits:
* Type-safe
* Mockable
* Allows to revert a temporary test workaround

### CODE REVIEW

File: src/net_processing.cpp [hunk 1]
```diff
@@ -4335,7 +4337,7 @@ void PeerManagerImpl::CheckForStaleTipAndEvictPeers()
 
     int64_t time_in_seconds = GetTime();
```
Comment (shaavan) [inline src/net_processing.cpp:4338]:
```suggestion
    auto time_in_seconds = GetTime<std::chrono::seconds>();
    
    EvictExtraOutboundPeers(time_in_seconds);
```

**Review by shaavan — COMMENTED** [review]
Concept ACK

- Used type-safe **std::chrono::seconds** to ensure that the compiler will validate types while compiling and throw an error if tried to assign the wrong type to a variable.
- Checked that the **std::chrono** is introduced correctly in the first commit.
- The second commit uses **std::chrono::seconds** variable `time_init` and `time_later` to `SetMockTime` in `src/test/denialofservices_tests.cpp` file. I have checked that these variables are used to `SetMockTime` at appropriate places.
However, I am not proficient enough (yet) to comment on if this fixes the bug that was initially described in #23739
- The third commit renames the three variables touched in this PR and names them to our latest naming convention.

I have some suggestions:

- Since the file `src/test/net_peer_eviction_tests.cpp` is touched in this PR, how about adding the clang formatting changes suggested by :arrow_down_small:  to this PR itself.
```
git diff -U0 HEAD~1.. | ./contrib/devtools/clang-format-diff.py -p1 -i -v
```

File: src/net_processing.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/net_processing.cpp:4338]: The diff as suggested wouldn't compile and needs more changes, so I'll leave it for a follow-up.

**Comment by MarcoFalke** [comment]
>     Since the file src/test/net_peer_eviction_tests.cpp is touched in this PR, how about adding the clang formatting changes suggested by arrow_down_small to this PR itself.

Looks like this will cause re-formats of lines in the file I haven't touched? Also, I think the file will/might be touched soon by other PRs, so I want to minimize the conflicts for now. Maybe it can be formatted in a pull that touches those lines directly?

**Comment by naumenkogs** [comment]
ACK fad943821e35d0eb2143061e718f0193e12a4c71

**Comment by shaavan** [comment]
ACK fad943821e35d0eb2143061e718f0193e12a4c71

> The diff as suggested wouldn't compile and needs more changes, so I'll leave it for a follow-up.

I wasn't aware of this. How about leaving a comment above this line explaining the needed follow-up.

> Maybe it can be formatted in a pull that touches those lines directly?

Formatting can be taken up as a direct follow-up to this PR as soon as this one gets merged.

**Comment by MarcoFalke** [comment]
> I wasn't aware of this. How about leaving a comment above this line explaining the needed follow-up.

Personally I am not a fan of adding comments to source code to list future refactoring that can be done. I think if it is worth it, it should be done in a pull request (or an issue created to track it).

Do you want to work on that?

**Comment by shaavan** [comment]
> Do you want to work on that?

I think I can give it a try!

**Comment by rebroad** [comment]
These variable rename pull requests are a pain in the ass - I thought it had been agreed previously that PRs should not be created to rename variables?

**Comment by sipa** [comment]
The purpose of this PR is not renaming variables; the point is moving to more type-safe representation of time.

When code is being changed already, the developer guidelines state that the new code should follow the style guidelines, and that's what is happening here.

Furthermore, in this specific case it also makes the change more reviewable. By changing the name, you know every access site is updated.

**Pull request merged**